enhancements/agent-installer/development-environment.md

```diff
@@ -0,0 +1,221 @@
+---
+title: ephemeral-agent-installer-e2e-tests
+authors:
+  - "@lranjbar"
+reviewers:
+  - "@andfasano"
+  - "@bfournie"
+  - "@hardys"
+  - "@pawanpinjarkar"
+  - "@rwsu"
+  - "@zaneb"
+approvers:
+  - "@dhellmann"
+  - "@celebdor"
+api-approvers: # in case of new or modified APIs or API extensions (CRD, aggregated api servers, webhooks, finalizers)
+  - N/A
+creation-date: 2022-02-10
+last-updated: 2022-03-29
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/AGENT-20
+---
+
+# Ephemeral Agent Installer Development Infrastructure
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Create a development environment for the ephemeral agent based installer image in the 
+[dev-scripts](https://github.com/openshift-metal3/dev-scripts) repository and refactor 
+components as needed for the new project. In addition we will add base case e2e test flows
+integrated into Openshift CI to get the project started.
+
+## Motivation
+
+The new agent based installer project will need a standard development environment for
+the team to use. We will need an e2e testing framework for validation in Openshift CI. 
+In this enhancement we will explain our desire to reuse existing frameworks for the new 
+project while limiting the scope of modifications to them to the new project's requirements.
+
+### Goals
+
+* Create a development environment for the agent installer project that can be used for local
+development and for automating the flows in Openshift CI.
+* Create predefined configurations to deploy Openshift in HA, compact and SNO topologies.
+* Create predefined network configurations for Openshift including but not limited IP type 
+(IPv4, IPv6, dual-stack), L2 segments and DHCP configurations.
```

Comment:
Will we say the enhancement isn't fully implemented until all of these profiles/configurations are available? Or is the goal to enable them to be created over time?

Reply:
My view is that we absolutely need this set because this is most the common configurations we use for OCP installs. The two that aren't: L2 segments and DHCP are linked in with work that is planned for Agent Installer. So it is fair to say we aren't complete until we get these in.

Reply:
Yeah, I'm not questioning whether need them. I'm questioning whether we would say "we have neither a development environment tool nor CI" until those are written.

It's safe to break this work down into more steps, and for this enhancement to say the goal is to provide 1 basic configuration and then enable us to build those "advanced" configurations later. You wouldn't need separate enhancements for the later work, but you might track them in separate tickets.

Reply:
Ahh okay I understand the question now. 🤔 I'll split them into different phases. It isn't too hard to do most of them because a lot of this already defined in Openshift CI

Reply:
Yeah that's a good point with disconnected vs connected the changes between the two are pretty advanced configurations 🤔